velox/functions/sparksql/Arithmetic.h

```diff
@@ -31,13 +31,27 @@ namespace facebook::velox::functions::sparksql {
 // The abs implementation is used for primitive types except for decimal type.
 template <typename TExec>
 struct AbsFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  bool ansiEnabled_ = false;
+
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const TInput* /*a*/) {
+    ansiEnabled_ = config.sparkAnsiEnabled();
+  }
+
   template <typename T>
   FOLLY_ALWAYS_INLINE void call(T& result, const T& a) {
     if constexpr (std::is_integral_v<T>) {
       if (FOLLY_UNLIKELY(a == std::numeric_limits<T>::min())) {
-        // To be compatible with Spark's ANSI off mode, when the input is
-        // negative minimum value, returns the same value as input instead of
-        // throwing an error.
+        if (ansiEnabled_) {
+          // In ANSI mode, throws an overflow error
+          VELOX_USER_FAIL("Arithmetic overflow: abs({})", a);
+        }
+        // In ANSI off mode, returns the same negative minimum value
```

Comment:
Please add a period at the end of each complete sentence in comments. Ditto for other comments.